src/brs/BlockchainProcessorImpl.java

```diff
@@ -1045,10 +1044,14 @@ private void pushBlock(final Block block) throws BlockNotAcceptedException {
 
         blockService.setPrevious(block, previousLastBlock);
         blockListeners.notify(block, Event.BEFORE_BLOCK_ACCEPT);
-        transactionProcessor.removeForgedTransactions(block.getTransactions());
+        transactionProcessor.removeForgedTransactions(transactions);
         transactionProcessor.requeueAllUnconfirmedTransactions();
         accountService.flushAccountTable();
         addBlock(block);
+        if(indirectIncomingService.isEnabled()){
```

Comment:
Moved this to a place _after_ transactions are stored in the database. This way we can take advantage of using foreign keys for referential integrity and cascaded deletion on indirect incomings on popoffs